nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiServiceFacade.java

```diff
@@ -6233,27 +6239,54 @@ protected Collection<AbstractMetricsRegistry> populateFlowMetrics() {
 
         // Get Connection Status Analytics (predictions, e.g.)
         Set<Connection> connections = controllerFacade.getFlowManager().findAllConnections();
-        for (Connection c : connections) {
-            // If a ResourceNotFoundException is thrown, analytics hasn't been enabled
+        Collection<Map<String, Long>> predictions = Collections.synchronizedList(new ArrayList<>());
+        final boolean analyticsEnabled = Boolean.parseBoolean(properties.getProperty(NiFiProperties.ANALYTICS_PREDICTION_ENABLED, Boolean.FALSE.toString()));
+
+        if (analyticsEnabled) {
+            if (parallelProcessingExecutor == null) {
+                parallelProcessingExecutor = Executors.newVirtualThreadPerTaskExecutor();
+            }
```

Comment:
This conditional instantiation is not thread safe for multiple invocations of the same method. The property needs to be checked earlier in the process.